src/supabase/queries/auth.ts

```diff
@@ -57,3 +51,45 @@ export const signOutUser = async () => {
     throw error;
   }
 };
+
+export const deleteCurrentUser = async (userId: string) => {
+  try {
+    await supabaseAdmin.auth.admin.deleteUser(userId);
+  } catch (error) {
+    throw error;
+  }
+};
+
+export const getCurrentUserInfo = async () => {
```

Comment:
We should define the return type to be `Promise<User>` but since `user` can be `User | null`, let's throw an error if it is null - since it shouldn't be. Refer to `getCaseById` in `cases.ts` for inspiration!